source/php/Component/Segment/segment.blade.php

```diff
@@ -4,6 +4,10 @@
         <div class="{{$baseClass}}__image {{ $imageClass }}" style="{{ $imageStyleString }}"></div>
     @endif
 
+    @if($layout == 'full-width' && $title||$content)
```

Comment:
Move to controller. Create a var named $hasOverlay. 

Curious why not a psuedoelement on main wrapper?